openapi2/openapi2.go

```diff
@@ -100,6 +103,54 @@ type (
 	}
 )
 
+func (o *Operation) toMap() (map[string]interface{}, error) {
+	type Alias Operation
+	data, err := json.Marshal((*Alias)(o))
+	if err != nil {
+		return nil, fmt.Errorf("json marshal: %v", err)
+	}
+
+	m := map[string]interface{}{}
+	if err := json.Unmarshal(data, &m); err != nil {
+		return nil, fmt.Errorf("json unmarshal: %v", err)
+	}
+
+	if o.Extend != nil {
+		if err := mergo.Merge(&m, o.Extend, mergo.WithOverride); err != nil {
+			return nil, fmt.Errorf("merge extend: %v", err)
+		}
+	}
+	return m, nil
+}
+
+// MarshalJSON implements the json.Marshaller interface.
+func (o *Operation) MarshalJSON() ([]byte, error) {
+	if o.Extend == nil {
+		type Alias Operation
```

Comment:
I'm not sure if this is needed but would be a nice some clarification via a code comment otherwise :thinking:

Reply:
Otherwise it'll marshal json will endlessly call this method,the new Alias type doesn't have the same methods so its safe to call